addons/web_editor/static/src/js/editor/image_processing.js

```diff
@@ -276,6 +285,29 @@ function loadImage(src, img = new Image()) {
 // want to load big images over the network many times when adjusting quality
 // and filter, we create a local cache of the images using object URLs.
 const imageCache = new Map();
+/**
+ * Loads the image into cache if not already and returns the object URL / dataURL.
+ *
+ * @param {string} src
+ */
+async function _loadImageData(src, loadDataURL = false) {
+    if (!imageCache.has(src) || (loadDataURL && !imageCache.get(src).dataURL)) {
+        const blob = await fetch(src).then(res => res.blob());
+        _updateImageData(src, {
+            objectURL: URL.createObjectURL(blob),
+            dataURL: loadDataURL ? await createDataURL(blob) : null,
+        });
+    }
+    return imageCache.get(src);
```

Comment:
I am not sure what is an object URL and data URL exactly but the implementation seems strange: we have a cache containing both but when we ask for `dataURL` we generate both but only use `dataURL`. I think this would be clearer if this was two independant cache, with two separate methods with explicit names (I don't see a cache where you call the method and want to use both `objectURL` and `dataURL`, right?

Reply:
Actually, using object URL makes things faster since they're just providing a blob local URL (no need to read the file and encode it as base64 dataURL). But unfortunately, they can't be used on shapes SVGs (Object URLs have a lifetime tied to current `document`...).

I think besides being not clear, this first implementation generates a new ObjectURL when we update dataURL for GIFs :grimacing:

I updated the code with separate methods to change only the targeted image DATA... :+1: